openedx/core/djangoapps/course_groups/models.py

```diff
@@ -49,3 +66,48 @@ class CourseUserGroupPartitionGroup(models.Model):
     )
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class CourseCohortsSettings(models.Model):
+    """
+    This model represents cohort settings for courses.
+    """
+    is_cohorted = models.BooleanField(default=False)
+
+    course_id = CourseKeyField(
+        unique=True,
+        max_length=255,
+        db_index=True,
+        help_text="Which course are these settings associated with?",
+    )
+
+    cohorted_discussions = models.TextField(null=True, blank=True)  # JSON list
+
+    # pylint: disable=invalid-name
+    always_cohort_inline_discussions = models.BooleanField(default=True)
+
+
+class CourseCohort(models.Model):
+    """
+    This model represents cohort related info.
+    """
+    course_user_group = models.OneToOneField(CourseUserGroup, unique=True, related_name='cohort')
+
+    RANDOM = 'random'
+    MANUAL = 'manual'
+    ASSIGNMENT_TYPE_CHOICES = ((RANDOM, 'Random'), (MANUAL, 'Manual'),)
+    assignment_type = models.CharField(max_length=20, choices=ASSIGNMENT_TYPE_CHOICES, default=MANUAL)
+
+    @classmethod
+    def create(cls, course_user_group, assignment_type=MANUAL):
+        """
+        Create a new course cohort info object.
+
+        Args:
+            course_user_group: CourseUserGroup
+            assignment_type: 'random' or 'manual'
+        """
+        return cls.objects.get_or_create(
+            course_user_group=course_user_group,
+            defaults={'assignment_type': assignment_type}
+        )
```

Comment:
any reason not to put the call to `CourseUserGroup.create` inside this method?  That was it only takes one call to create both CUG and CC rows and make a 'complete' cohort.